### ISSUE TITLE
The EM algorithm

### ISSUE THREAD

**Issue by vdv1g16** [tensorflow/ranking#66]
In the paper The LambdaLoss Framework for Ranking Metric Optimization, the loss is optimized by EM algorithm. However, I don't see the EM algorithm anywhere in the tf ranking code. 
Please explain. Many thanks

**Comment by xuanhuiwang** [comment]
In TensorFlow, we have minibatches and the EM is very much approximated in our library.

Before each minibatch we have the old parameter \Theta. It is used to compute the scores and ranks for the minibatch. Then we compute the losses for the minibatch. The optimization of the loss is done by the mini-batch gradient descent and back propagation. After the back propagation, we have a new \Theta.

This is different from the standard EM and in some sense, it is a stochastic version of Generalized EM.
